crates/noirc_evaluator/src/ssa/acir_gen.rs

```diff
@@ -630,6 +638,38 @@ pub fn evaluate_cmp(
     }
 }
 
+//Performs byte decomposition
+pub fn to_bytes(lhs: &InternalVar, bit_size: u32, evaluator: &mut Evaluator) -> Vec<Witness> {
+    assert!(bit_size < FieldElement::max_num_bits());
+    let mut bytes = Expression::default();
+    let mut two_pow = FieldElement::one();
+    let two = FieldElement::from(2_i128);
+    let mut result = Vec::new();
+    for _ in 0..(bit_size / 8) {
+        let byte_witness = evaluator.add_witness_to_cs();
+        result.push(byte_witness);
+        let byte_expr = from_witness(byte_witness);
+        bytes = add(&bytes, two_pow, &byte_expr);
+        for _ in 0..8 {
+            two_pow = two_pow.mul(two);
+        }
+        evaluator.gates.push(Gate::Arithmetic(subtract(
```

Comment:
why do you add this constraint which does not constrain anything?

Reply:
I actually wanted to ask about this. I basically wanted to constrain the byte expr I just created, however, I was wondering if I have to make it quadratic. Such as mul(byte_expr, byte_expr) - mul(byte_expr, byte_expr), or if this constraint is even required?

Reply:
you want to constraint it to what?

Reply:
I was originally thinking we had to constrain the intermediate decomposed expression, but I see how the constraint is kind of pointless now. 

Under the `split` function am I correct in assuming that this [line](https://github.com/noir-lang/noir/blob/dff8f393706c102055e78b4d97df207d73640eee/crates/noirc_evaluator/src/ssa/acir_gen.rs#L646) constrains the bit expression to be either a 1 or 0. I just tried to generalize `split` and `to_bytes` though with `to_radix` and I am now wondering if it is dangerous to remove this?

Reply:
I ended up using the `range_constraint` function already in the `acir_gen.rs` as I realized this is what I wanted. This is sufficient for constraining the generalized functionality of `split` and `to_bytes`.